magda-web-server/src/buildSitemapRouter.ts

```diff
@@ -0,0 +1,123 @@
+import * as express from "express";
+import Registry from "@magda/typescript-common/dist/registry/RegistryClient";
+const sm = require("sitemap");
+
+const DATASET_REQUIRED_ASPECTS = ["dcat-dataset-strings"];
+
+export type SitemapRouterOptions = {
+    baseExternalUrl: string;
+    registry: Registry;
+};
+
+export default function buildSitemapRouter({
+    baseExternalUrl,
+    registry
+}: SitemapRouterOptions): express.Router {
+    const app = express();
+
+    // Make sure every request is setting XML as the content type.
+    app.use((req, res, next) => {
+        res.header("Content-Type", "application/xml");
+        next();
+    });
+
+    app.get("/", (req, res) => {
+        catchError(
+            res,
+            registry
+                .getRecordsPageTokens(DATASET_REQUIRED_ASPECTS)
+                .then(handleError)
+                .then(async result => {
+                    const datasetsPages = result.map(token => {
+                        return (
+                            baseExternalUrl +
+                            "/sitemap/dataset/afterToken/" +
```

Comment:
Use urijs to build the URL so that it's not sensitive to whether or not `baseExternalUrl` has a trailing slash.